pkg/xray/resource_xray_license_policy.go

```diff
@@ -20,143 +21,177 @@ func resourceXrayLicensePolicyV2() *schema.Resource {
+				ForceNew:    true,
+				Description: "Name of the policy (must be unique)",
 			},
 			"description": {
-				Type:     schema.TypeString,
-				Optional: true,
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "More verbose description of the policy",
 			},
 			"type": {
-				Type:     schema.TypeString,
-				Required: true,
+				Type:             schema.TypeString,
+				Required:         true,
+				Description:      "Type of the policy",
+				ValidateDiagFunc: inList("Security", "License"),
 			},
 			"author": {
-				Type:     schema.TypeString,
-				Computed: true,
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "User, who created the policy",
 			},
 			"created": {
-				Type:     schema.TypeString,
-				Computed: true,
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Creation timestamp",
 			},
 			"modified": {
-				Type:     schema.TypeString,
-				Computed: true,
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Modification timestamp",
 			},
 			"rules": {
-				Type:     schema.TypeList,
-				Required: true,
+				Type:        schema.TypeList,
+				Required:    true,
+				Description: "Nested block describing the policy rules. Described below",
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
 						"name": {
-							Type:     schema.TypeString,
-							Required: true,
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "Name of the rule",
 						},
 						"priority": {
-							Type:     schema.TypeInt,
-							Required: true,
+							Type:             schema.TypeInt,
+							Required:         true,
+							ValidateDiagFunc: validation.ToDiagFunc(validation.IntAtLeast(1)),
+							Description:      "Integer describing the rule priority. Must be at least 1",
 						},
 						"criteria": {
-							Type:     schema.TypeList,
-							Required: true,
-							MinItems: 1,
-							MaxItems: 1,
+							Type:        schema.TypeSet,
+							Required:    true,
+							Description: "Nested block describing the criteria for the policy. Described below",
+							MinItems:    1,
 							Elem: &schema.Resource{
 								Schema: map[string]*schema.Schema{
 									"banned_licenses": {
-										Type:     schema.TypeList,
-										Optional: true,
+										Type:        schema.TypeSet,
+										Optional:    true,
+										Description: "A list of OSS license names that may not be attached to a component.",
 										Elem: &schema.Schema{
-											Type: schema.TypeString,
+											Type:         schema.TypeString,
+											ValidateFunc: licenseTypeValidator,
 										},
 									},
 									"allowed_licenses": {
-										Type:     schema.TypeList,
-										Optional: true,
+										Type:        schema.TypeSet,
+										Optional:    true,
+										Description: "A list of OSS license names that may be attached to a component.",
```

Comment:
Should we use `ConflictsWith` here with `banned_licenses` as a early validation?

Reply:
I've tried it initially and now again: `        1 error occurred:
        	* resource xray_license_policy: ConflictsWith: banned_licenses references
        unknown attribute (allowed_licenses) at part (allowed_licenses)`
Not sure why it doesn't work, because I see similar code in Artifactory provider

Reply:
Looks like it isn't supported at the moment: https://github.com/hashicorp/terraform-plugin-sdk/issues/71#issuecomment-666613275